src/Symfony/Component/DependencyInjection/Reference.php

```diff
@@ -29,7 +29,7 @@ class Reference
      */
     public function __construct($id, $invalidBehavior = ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE)
     {
-        $this->id = strtolower($id);
+        $this->id = $id;
```

Comment:
we still need to cast to string here (with a deprecation), otherwise ``__toString`` will be broken